tools/cluster/tests/cluster_stability_test.go

```diff
@@ -0,0 +1,574 @@
+/**
+This test will test the recovery capabilities of the clusterized nodes, where some nodes can break/restart/lay dead at any time.
+*/
+
+package tests
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/iotaledger/wasp/contracts/native/inccounter"
+	"github.com/stretchr/testify/require"
+)
+
+type SabotageEnv struct {
+	chainEnv      *chainEnv
+	NumValidators int
+	From          int
+	To            int
+	SabotageList  []int
+}
+
+func InitializeStabilityTest(t *testing.T, numValidators, clusterSize int) *SabotageEnv {
+	progHash := inccounter.Contract.ProgramHash
+	env := setupWithChain(t, clusterSize)
+	_, _, err := env.clu.InitDKG(numValidators)
+
+	require.NoError(t, err)
+
+	_, _ = env.chain.DeployContract(incCounterSCName, progHash.String(), "testing with inccounter", nil)
+	waitUntil(t, env.contractIsDeployed(incCounterSCName), env.clu.Config.AllNodes(), 50*time.Second, "contract is deployed")
+
+	return &SabotageEnv{
+		chainEnv:      env,
+		NumValidators: numValidators,
+		From:          0,
+		To:            0,
+		SabotageList:  make([]int, 0),
+	}
+}
+
+func (e *SabotageEnv) sendRequests(numRequests int, messageDelay time.Duration) {
+	for i := 0; i < numRequests; i++ {
+		client := e.chainEnv.createNewClient()
+
+		_, err := client.PostRequest(inccounter.FuncIncCounter.Name)
+		require.NoError(e.chainEnv.t, err)
+
+		time.Sleep(messageDelay)
+	}
+}
+
+func (e *SabotageEnv) setSabotageValidators(breakCount int) {
+	clusterSize := e.chainEnv.clu.Config.Wasp.NumNodes
+	nodeList := []int{}
+
+	from := clusterSize - e.NumValidators
+	to := from + breakCount - 1
+
+	for i := from; i <= to; i++ {
+		nodeList = append(nodeList, i)
+	}
+
+	e.From = from
+	e.To = to
+	e.SabotageList = nodeList
+}
+
+func (e *SabotageEnv) setSabotageAll(breakCount int) {
+	nodeList := []int{}
+
+	from := 1
+	to := from + breakCount - 1
+
+	for i := from; i <= to; i++ {
+		nodeList = append(nodeList, i)
+	}
+
+	e.From = from
+	e.To = to
+	e.SabotageList = nodeList
+}
+
+type SabotageOption int
+
+const (
+	SabotageByKill SabotageOption = iota
+	// Important: Frozen nodes need to get killed/unfrozen manually after the test is done, otherwise they stay alive after the test has finished
+	SabotageByFreeze SabotageOption = iota
+)
```

Comment:
```suggestion
const (
	SabotageByKill SabotageOption = iota
	// Important: Frozen nodes need to get killed/unfrozen manually after the test is done, otherwise they stay alive after the test has finished
	SabotageByFreeze SabotageOption
)
```

both have the same value, right? 😅

Reply:
No not right :) 
You can simplify it by leaving the second iota out. But the result is the same. 

https://yourbasic.org/golang/iota/